x-pack/plugins/actions/server/builtin_action_types/lib/connector_token_client.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ConnectorToken } from '../../types';
+import { EncryptedSavedObjectsClient } from '../../../../encrypted_saved_objects/server';
+import { SavedObjectsClientContract, SavedObjectsUtils } from '../../../../../../src/core/server';
+import { CONNECTOR_TOKEN_SAVED_OBJECT_TYPE } from '../../constants/saved_objects';
+
+// We are expecting max possible token types for the single connector to be not bigger then 2 - for access and refresh tokens
+export const MAX_TOKENS_RETURNED = 2;
```

Comment:
Would it be possible to run into a race condition where multiple actions using the same connector are being executed at the same time and they both request a token and create two access token saved objects for the same connector?

Reply:
Good call here. I think we might better to follow the approach as we do for update method for ActionsClient - use create with overwrite. 
Also, it will be good to add here createdAt time, to be (in the client get method) able to sort and fetch the most recently created(updated) token if the race condition happens.